streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -192,17 +224,15 @@ public KafkaStreams(final TopologyBuilder builder, final StreamsConfig config, f
     public synchronized void start() {
         log.debug("Starting Kafka Stream process");
 
-        if (state == CREATED) {
+        if (state == KafkaStreamsState.CREATED) {
             for (final StreamThread thread : threads)
                 thread.start();
 
-            state = RUNNING;
+            state = KafkaStreamsState.RUNNING;
 
             log.info("Started Kafka Stream process");
-        } else if (state == RUNNING) {
-            throw new IllegalStateException("This process was already started.");
```

Comment:
Any particular reason for removing this? I guess, there was a reason to put it in the first place. But to be honest, I don't see the reason myself. Start() can just be idempotent.